components/benefit_sponsors/app/models/benefit_sponsors/benefit_sponsorships/benefit_sponsorship.rb

```diff
@@ -582,11 +582,12 @@ def application_event_subscriber(aasm)
       when :expired
         cancel! if may_cancel?
       when :canceled
-        cancel! if (may_cancel? && aasm.current_event == :activate_enrollment!)
+        if aasm.current_event == :activate_enrollment! || aasm.from_state == :enrollment_ineligible
```

Comment:
cancel benefit sponsorship to move to applicant state when benefit application cancelled.